### PR TITLE
Gen3 animations

### PR DESCRIPTION
**Pull request by 8bitZeta**
Finally ready to merge the Gen 3 Move Animations.

### CODE REVIEW

File: data/moves/animation_pointers.asm [hunk 1]
```diff
@@ -3,22 +3,22 @@ BattleAnimations::
...
 	indirect_entries GEN1_MOVES, BattleAnimationsGen1
 	indirect_entries GEN2_MOVES, BattleAnimationsGen2
+	indirect_entries GEN3_MOVES, BattleAnimationsGen3
 	indirect_entries $ffff - NUM_BATTLE_ANIMS
 	indirect_entries $ffff, BattleAnimationsNegatives
 	indirect_table_end
 
 
 SECTION "Battle Animations Gen 1", ROMX
-
```
Comment (vulcandth) [inline data/moves/animation_pointers.asm]: Don't delete this space.

File: data/moves/animation_pointers.asm [hunk 2]
```diff
@@ -3,22 +3,22 @@ BattleAnimations::
...
 INCLUDE "data/moves/animations_gen1.asm"
 
-
```
Comment (vulcandth) [inline data/moves/animation_pointers.asm]: Don't delete this space.

File: data/moves/animation_pointers.asm [hunk 3]
```diff
@@ -3,22 +3,22 @@ BattleAnimations::
...
 SECTION "Battle Animations Gen 2", ROMX
-
```
Comment (vulcandth) [inline data/moves/animation_pointers.asm:17]: Don't delete this space.

File: data/moves/animation_pointers.asm [hunk 4]
```diff
@@ -3,22 +3,22 @@ BattleAnimations::
...
 INCLUDE "data/moves/animations_gen2.asm"
 
-SECTION "Battle Animations Gen Negatives", ROMX
+SECTION "Battle Animations Gen 3", ROMX
+INCLUDE "data/moves/animations_gen3.asm"
 
+SECTION "Battle Animations Gen Negatives", ROMX
 INCLUDE "data/moves/animations_negatives.asm"
```
Comment (vulcandth) [inline data/moves/animation_pointers.asm]:
```suggestion

SECTION "Battle Animations Gen 3", ROMX

INCLUDE "data/moves/animations_gen3.asm"


SECTION "Battle Animations Gen Negatives", ROMX

INCLUDE "data/moves/animations_negatives.asm"
```

File: data/moves/descriptions.asm [hunk 5]
```diff
@@ -260,8 +261,114 @@ MoveDescriptionsGen2:
 	dw RockSmashDescription
 	dw WhirlpoolDescription
 	dw BeatUpDescription
+	.IndirectEnd::
```
Comment (vulcandth) [inline data/moves/descriptions.asm]:
```suggestion
.IndirectEnd::
```

File: data/moves/descriptions.asm [hunk 6]
```diff
@@ -260,8 +261,114 @@ MoveDescriptionsGen2:
...
+	
+	MoveDescriptionsGen3:
```
Comment (vulcandth) [inline data/moves/descriptions.asm]:
```suggestion
MoveDescriptionsGen3:
```

File: data/moves/effects.asm [hunk 7]
```diff
@@ -2072,6 +2072,16 @@ DefenseCurl:
...
+;	New Moves Start Here
+;==========================
+Hail:
+	checkobedience
+	usedmovetext
+	doturn
+	starthail
+	endmove
 
 LowKick:
```
Comment (vulcandth) [inline data/moves/effects.asm:2083]: LowKick isn't a new move...

File: data/moves/effects_pointers.asm [hunk 8]
```diff
@@ -158,5 +158,10 @@ MoveEffectsPointers:
 	dw BeatUp
 	dw Fly
 	dw DefenseCurl
+	;==========================
+	;	New Moves Start Here
+	;==========================
+	dw FreezeHit ; for Blizzard, purposefully with different EFFECT_* constant
+	dw Hail
 	dw LowKick
```
Comment (vulcandth) [inline data/moves/effects_pointers.asm:163]: Lowkick isn't a new move

File: macros/scripts/battle_commands.asm [hunk 9]
```diff
@@ -180,7 +180,11 @@ ENDM
 	command supereffectivelooptext  ; ad
 	command startloop               ; ae
 	command curl                    ; af
-	command lowkick                 ; b0
+	;==========================
+	;	New Moves Start Here
+	;==========================
+	command starthail               ; b0
+	command lowkick                 ; b1
```
Comment (vulcandth) [inline macros/scripts/battle_commands.asm:184]: Again; lowkick isn't a new move.

File: constants/move_effect_constants.asm [hunk 10]
```diff
@@ -157,5 +157,10 @@
 	const EFFECT_BEAT_UP
 	const EFFECT_FLY
 	const EFFECT_DEFENSE_CURL
+	;==========================
+	;	New Moves Start Here
+	;==========================
+	const EFFECT_BLIZZARD
+	const EFFECT_HAIL
 	const EFFECT_LOW_KICK
```
Comment (vulcandth) [inline constants/move_effect_constants.asm:162]: Low kick not a new move

File: data/battle/effect_command_pointers.asm [hunk 11]
```diff
@@ -180,5 +180,9 @@ BattleCommandPointers:
 	dw BattleCommand_SuperEffectiveLoopText
 	dw BattleCommand_StartLoop
 	dw BattleCommand_Curl
+	;==========================
+	;	New Moves Start Here
+	;==========================
+	dw BattleCommand_StartHail
 	dw BattleCommand_LowKick
```
Comment (vulcandth) [inline data/battle/effect_command_pointers.asm:184]: lowkick not a new move

**Review by vulcandth — CHANGES_REQUESTED** [review]
Thanks for all the work! Just a couple quick minor things

File: data/moves/descriptions.asm (hunk 5 above)
Comment (8bitZeta) [reply data/moves/descriptions.asm]: Oh that's just gross. Completely overlooked this.

File: data/moves/animation_pointers.asm (hunk 4 above)
Comment (8bitZeta) [reply data/moves/animation_pointers.asm]: It has been done. Wasn't sure on the formatting is all.

File: data/battle/effect_command_pointers.asm (hunk 11 above)
Comment (8bitZeta) [reply data/battle/effect_command_pointers.asm:184]: Would changing the wording to "New Effects Start Here" be sufficient? I believe that's what initially intended.

Comment (vulcandth) [reply data/battle/effect_command_pointers.asm:184]: Honestly... i'd just remove the comments altogether... when we start gen4.. would we have.. `new new moves start here`?

Doesn't really serve a purpose imo... and can be confusing.

Comment (8bitZeta) [reply data/battle/effect_command_pointers.asm:184]: It was just to separate vanilla is all, but I can go and remove them. I'm not married to them.

Comment (8bitZeta) [reply data/battle/effect_command_pointers.asm:184]: Comments have been removed.

**Pull request merged**